src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -20418,24 +20424,37 @@ bool Compiler::fgDumpFlowGraph(Phases phase)
     {
         if (createDotFile)
         {
-            // Add constraint edges to try to keep nodes ordered.
-            // It seems to work best if these edges are all created first.
-            switch (block->bbJumpKind)
+            fprintf(fgxFile, "    BB%02u [label = \"BB%02u\\n\\n", block->bbNum, block->bbNum);
+
+            // "Raw" Profile weight
+            if (block->hasProfileWeight())
             {
-                case BBJ_COND:
-                case BBJ_NONE:
-                    assert(block->bbNext != nullptr);
-                    fprintf(fgxFile, "    " FMT_BB " -> " FMT_BB "\n", block->bbNum, block->bbNext->bbNum);
-                    break;
-                default:
-                    // These may or may not have an edge to the next block.
-                    // Add a transparent edge to keep nodes ordered.
```

Comment:
Although I realize that it may not always be the best layout, I find it *so* much easier to reason about these graphs when the nodes are in sequential order by `bbNum`. IIRC this was an imperfect mechanism for doing so, but it mostly worked.

Reply:
So you'd like to see a spine of blocks in increasing bbNum order? I can add the invisible edges back.

Reply:
Yes, that would be my preference, unless there's consensus against it.

Reply:
Internal blocks like a loop preheader block have a high number at the are allocated as the highest block number.  (Until we renumber the blocks)

Reply:
My preference is to have the fall through block alwasy be a short straight line down to the next block:

block A
    |
    |
block B

Reply:
I can do invisible edges to encourage `bbNext` order, that will give `bbNum` order if you've recently renumbered, and `bbNext` order otherwise.

Reply:
That sounds great to me.